src/wix/WixToolset.Core.Burn/Bundles/BurnWriter.cs

```diff
@@ -138,10 +143,19 @@ public bool AppendContainer(Stream containerStream, long containerSize, BurnComm
                     break;
 
                 case Container.Attached:
-                    burnSectionCount = 2;
-                    burnSectionOffsetSize = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE;
+                    burnSectionCount = 2 + (uint)this.AttachedContainers.Count;
```

Comment:
The hardcoded 2 was for UX container + attached container. This needs to be `1 + (uint)this.AttachedContainers.Count`.